File: app/src/main/java/dev/radley/omgstarwars/di/ApiComponent.java
package dev.radley.omgstarwars.di;


import dagger.Component;
import dev.radley.omgstarwars.network.StarWarsService;
import dev.radley.omgstarwars.viewmodels.CategoryViewModel;
import dev.radley.omgstarwars.viewmodels.DetailViewModel;
import dev.radley.omgstarwars.viewmodels.SearchViewModel;


@Component(modules = {ApiModule.class})
public interface ApiComponent {

    void inject(StarWarsService service);

    void inject(CategoryViewModel viewModel);

    void inject(DetailViewModel viewModel);

    void inject(SearchViewModel viewModel);
}

File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/dagger/ApiComponent.kt
package dev.radley.omgstarwars.dagger


import dagger.Component
import dev.radley.omgstarwars.network.StarWarsService
import dev.radley.omgstarwars.viewmodels.CategoryViewModel
import dev.radley.omgstarwars.viewmodels.DetailViewModel
import dev.radley.omgstarwars.viewmodels.SearchViewModel
import javax.inject.Singleton


@Component(modules = [ApiModule::class])
interface ApiComponent {

    @Singleton
    fun inject(service: StarWarsService)

    fun inject(viewModel: CategoryViewModel)

    fun inject(viewModel: DetailViewModel)

    fun inject(viewModel: SearchViewModel)
}
